packages/kit/src/runtime/server/cookie.js

```diff
@@ -1,48 +1,65 @@
 import * as cookie from 'cookie';
 
+/** @type {import('cookie').CookieSerializeOptions} */
+const DEFAULT_SERIALIZE_OPTIONS = {
+	httpOnly: true,
+	secure: true,
+	path: '/',
+	sameSite: 'lax'
+};
+
 /**
  * @param {Request} request
  * @param {URL} url
  */
 export function get_cookies(request, url) {
 	const initial_cookies = cookie.parse(request.headers.get('cookie') ?? '');
 
-	/** @type {Array<{ name: string, value: string, options: import('cookie').CookieSerializeOptions }>} */
-	const new_cookies = [];
+	/** @type {Map<string, {name: string; value: string; options: import('cookie').CookieSerializeOptions;}>} */
+	const new_cookies = new Map();
 
 	/** @type {import('types').Cookies} */
 	const cookies = {
 		get(name, opts) {
 			const decode = opts?.decode || decodeURIComponent;
-
-			let i = new_cookies.length;
-			while (i--) {
-				const cookie = new_cookies[i];
-
-				if (
-					cookie.name === name &&
-					domain_matches(url.hostname, cookie.options.domain) &&
-					path_matches(url.pathname, cookie.options.path)
-				) {
-					return cookie.value;
-				}
+			const c = new_cookies.get(name);
+			if (
+				c &&
+				domain_matches(url.hostname, c.options.domain) &&
+				path_matches(url.pathname, c.options.path)
+			) {
+				return c.value;
 			}
 
 			return name in initial_cookies ? decode(initial_cookies[name]) : undefined;
```

Comment:
There's a gotcha here. `initial_cookies` have already been decoded by `cookie.parse`, using the default `decodeURIComponent`.  Decoding twice is likely bad. Two options:

1. Parse the cookies each time inside `get`, using the passed `decode` option if present.
2. Or, we just say that you cannot provide your own `decode` and `encode` options.  That'd be my pref, though there's gonna be somebody who won't like it. 

Not sure what to do. I'll do "parse inside get" for now unless @benmccann @Rich-Harris have any thoughts.

Reply:
PR updated with "parse inside get".

Here's the test that would fail otherwise: https://github.com/sveltejs/kit/blob/a73cad0e36c944774c3c603fb1f5895d18c20962/packages/kit/test/apps/basics/test/test.js#L1849